### PR TITLE
Adventure fix

### PR DESCRIPTION
**Pull request by RKelln**
WIP: do not merge, just for showing my work.

### CODE REVIEW

File: serverjs/updatecards.js [hunk 1]
```diff
@@ -347,7 +347,8 @@ function convertColors(card, isExtra) {
   } else {
     if (!card.card_faces || card.layout == 'flip') {
       return colors.concat(card.colors);
-    } else if (card.layout == 'split') {
+    } else if (card.layout == 'split' || card.layout == 'adventure') {
+      if (card.name.startsWith('Flaxen')) console.log(card);
```
Comment (dekkerglen) [inline serverjs/updatecards.js]: remove debug line

File: serverjs/updatecards.js [hunk 2]
```diff
@@ -596,6 +597,9 @@ function convertCard(card, isExtra) {
   newcard.scryfall_uri = card.scryfall_uri;
   newcard.rarity = card.rarity;
   newcard.oracle_text = card.oracle_text;
+  if (card.layout == 'adventure') {
+    newcard.oracle_text = card.card_faces[0].oracle_text + '\n' + card.card_faces[1].oracle_text;
+  }
```
Comment (dekkerglen) [inline serverjs/updatecards.js]: can we do this for all cards with multiple 'faces' objects?

File: src/util/Filter.js [hunk 3]
```diff
@@ -446,35 +446,32 @@ function filterApply(card, filter, inCube) {
   if (filter.category == 'oracle' && card.details.oracle_text) {
     res = card.details.oracle_text.toLowerCase().indexOf(filter.arg) > -1;
   }
-  if (filter.category == 'color' && card.details.colors) {
+  if (filter.category == 'color' && colors !== undefined) {
```
Comment (dekkerglen) [inline src/util/Filter.js]: 'colors' is defined above as the card's color_identiy, or override. We should be using card.details.colors for the color filter

File: src/util/Filter.js [hunk 4]
```diff
@@ -695,14 +692,44 @@ function filterApply(card, filter, inCube) {
...
+          if (symbol.includes('-')) {
+            res = false;
+          }
+        });
+        break;
+
       case 'hybrid':
-      case 'phyrexian':
-          let type = filter.arg.substring(0,1).toUpperCase() + filter.arg.substring(1);
-          res = cardIsLabel(card, type, 'Manacost Type');
```
Comment (dekkerglen) [inline src/util/Filter.js]: why are we not using cardIsLabel for this? Shouldn't we be consistent?

Comment (RKelln) [reply src/util/Filter.js]: Yes, would prefer to reuse - but I got stuck trying to figure out how to require the file. This code is just for testing until we figure out how to best share code between client and server.

File: serverjs/updatecards.js (hunk 2 above)
Comment (RKelln) [reply serverjs/updatecards.js]: Yup, good idea (for searching purposes). Is the oracle text used for anything else that would muck up if we concatenate it?

Comment (dekkerglen) [reply serverjs/updatecards.js]: Nope, just used for filtering

File: src/util/Filter.js (hunk 4 above)
Comment (dekkerglen) [reply src/util/Filter.js]: Isn't this already included? It shows you've replaced this code with something else

**Comment by dekkerglen** [comment]
New PR for this fix has been created

**Pull request closed without merging**